forge-game/src/main/java/forge/game/player/Player.java

```diff
@@ -3116,7 +3153,72 @@ else if (game.getRules().hasAppliedVariant(GameType.Oathbreaker)) {
+        }
+    }
+    public void setRingLevel(int level) {
+        if (getTheRing() == null)
+            createTheRing(null);
+        if (level == 1) {
+            String legendary = "Mode$ Continuous | EffectZone$ Command | Affected$ Card.YouCtrl+IsRingbearer | AddType$ Legendary | Description$ Your Ring-bearer is legendary.";
+            String cantBeBlocked = "Mode$ CantBlockBy | EffectZone$ Command | ValidAttacker$ Card.YouCtrl+IsRingbearer | ValidBlockerRelative$ Creature.powerGTX | Description$ Your Ring-bearer can't be blocked by creatures with greater power.";
+            getTheRing().addStaticAbility(legendary);
+            StaticAbility st = getTheRing().addStaticAbility(cantBeBlocked);
+            st.setSVar("X", "Count$CardPower");
+        } else if (level == 2) {
+            final String attackTrig = "Mode$ Attacks | ValidCard$ Card.YouCtrl+IsRingbearer | TriggerDescription$ Whenever your ring-bearer attacks, draw a card, then discard a card. | TriggerZones$ Command";
+            final String drawEffect = "DB$ Draw | Defined$ You | NumCards$ 1";
+            final String discardEffect = "DB$ Discard | Defined$ You | NumCards$ 1 | Mode$ TgtChoose";
+
+            final Trigger attackTrigger = TriggerHandler.parseTrigger(attackTrig, getTheRing(), true);
+
+            SpellAbility drawExecute = AbilityFactory.getAbility(drawEffect, getTheRing());
+            AbilitySub discardExecute = (AbilitySub) AbilityFactory.getAbility(discardEffect, getTheRing());
 
+            drawExecute.setSubAbility(discardExecute);
+            attackTrigger.setOverridingAbility(drawExecute);
+            getTheRing().addTrigger(attackTrigger);
+        } else if (level == 3) {
+            final String becomesBlockedTrig = "Mode$ AttackerBlockedByCreature | ValidCard$ Card.YouCtrl+IsRingbearer| ValidBlocker$ Creature | TriggerZones$ Command | TriggerDescription$ Whenever your Ring-bearer becomes blocked a creature, that creature's controller sacrifices it at the end of combat.";
+            final String endOfCombatTrig = "DB$ DelayedTrigger | Mode$ Phase | Phase$ EndCombat | RememberObjects$ TriggeredBlockerLKICopy | TriggerDescription$ At end of combat, the controller of the creature that blocked CARDNAME sacrifices that creature.";
+            final String sacBlockerEffect = "DB$ Destroy | Defined$ DelayTriggerRememberedLKI | Sacrifice$ True";
+
+            final Trigger becomesBlockedTrigger = TriggerHandler.parseTrigger(becomesBlockedTrig, getTheRing(), true);
+
+            SpellAbility endCombatExecute = AbilityFactory.getAbility(endOfCombatTrig, getTheRing());
+            AbilitySub sacExecute = (AbilitySub) AbilityFactory.getAbility(sacBlockerEffect, getTheRing());
+
+            endCombatExecute.setAdditionalAbility("Execute", sacExecute);
+            becomesBlockedTrigger.setOverridingAbility(endCombatExecute);
+            getTheRing().addTrigger(becomesBlockedTrigger);
+        } else if (level >= 4) {
```

Comment:
this change seems wrong, now it would keep gaining this trigger

Reply:
thanks, anyway does the effect name can be localized safely?